pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,401 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+
+"""
+Base classes for an object store for CIM classes, CIM instances, and CIM
+qualifier declarations and the generic API for a repository of these object
+types organized by repository.
+"""
+
+from abc import abstractmethod, abstractproperty
+from six import add_metaclass
+from custom_inherit import DocInheritMeta
+
+import six
+
+
+def compatibleabstractproperty(func):
+    """
+    Python 2 and python 3 differ in decorator for abstract property.
+    in python 3 (gt 3.3) it is:
+        @property
+        @abstractproperty
+    in python 2
+        @abstractproperty
+    """
+
+    if six.PY2:  # pylint: disable=no-else-return
+        return abstractproperty(func)
+    else:
+        return property(abstractmethod(func))
+
+
+@add_metaclass(DocInheritMeta(style="google", abstract_base_class=True))
+class BaseObjectStore(object):
+    """
+    An abstract class that defines the APIs for the methods of an object store
+    for CIM objects including CIMClass, CIMInstance, and CIMQualifierDeclaration
+    objectsthat constitute a WBEM server repository.  This
+    class provides the abstract methods for creating, accessing, and deleting,
+    CIM objects of a single CIM object type in the repository.
+
+    These APIs allows creating, updating, deleting, and retrieving these
+    objects.
+    """
+
+    def __init__(self, cim_object_type):
+        """
+        Initialize the object store.
```

Comment:
The docstring of `__init__()` contains "Initialize the object store.". For consistency with my similar comment on the derived class, I suggest to remove this sentence here in the base class as well.

Reply:
DONE